### PR TITLE
fix(module:time-picker): placeholder not update with i18n changes

### PR DESCRIPTION
**Pull request by DepickereSven**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?
When you update the ``i18n.setLocale()`` after a few seconds the placeholder of the ``time-picker component`` will not update. 

Reproduce https://stackblitz.com/edit/angular-qyneqn?file=src/app/app.component.ts

## What is the new behavior?
Whenever you update ``i18n.setLocale()`` the placeholder will also update.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by DepickereSven** [comment]
@wenqi73 is it possible to review this PR because it's an annoying bug that the language isn't correct when you reload the page.

File: components/time-picker/time-picker.component.ts [hunk 1]
```diff
@@ -281,6 +285,11 @@ export class NzTimePickerComponent implements ControlValueAccessor, OnInit, Afte
   ngOnInit(): void {
     this.inputSize = Math.max(8, this.nzFormat.length) + 2;
     this.origin = new CdkOverlayOrigin(this.element);
+
+    this.i18nPlaceHolder$ = this.i18n.localeChange.pipe(
+      map((nzLocale: NzI18nInterface) => nzLocale.TimePicker.placeholder),
+      startWith(this.i18n.getLocaleData(`TimePicker.lang.placeholder`))
+    );
```
Comment (wenqi73) [inline components/time-picker/time-picker.component.ts:292]:
```suggestion
    this.i18nPlaceHolder$ = this.i18n.localeChange.pipe(
      map((nzLocale: NzI18nInterface) => nzLocale.TimePicker.placeholder)
    );
```
Already has the initial value, no need to use `startWith`.

**Review by wenqi73 — CHANGES_REQUESTED** [review]
Thanks for your contribution, add a unit test example would be better.

**Comment by DepickereSven** [comment]
@wenqi73 I added a test case for the change I made. 
But I had also a merge conflict to solve.
I resolved the merge conflict with a merge from master and now I have a lot of file changes.
But they are formatting changes, do I need to create a new PR or is it okay like this?

File: components/time-picker/time-picker.component.spec.ts [hunk 2]
```diff
@@ -166,20 +188,28 @@ describe('time-picker', () => {
...
   nzSuffixIcon?: string;
+  i18nPlaceHolder$: Observable<string | undefined> = of(undefined);
   onChange(): void {}
   @ViewChild(NzTimePickerComponent, { static: false }) nzTimePickerComponent!: NzTimePickerComponent;
+
+  constructor(private i18n: NzI18nService) {}
+
+  ngOnInit(): void {
+    this.i18nPlaceHolder$ = this.i18n.localeChange.pipe(map((nzLocale: NzI18nInterface) => nzLocale.TimePicker.placeholder));
+  }
```
Comment (wenqi73) [inline components/time-picker/time-picker.component.spec.ts:214]:
```suggestion

```

File: components/time-picker/time-picker.component.spec.ts [hunk 3]
```diff
@@ -166,20 +188,28 @@ describe('time-picker', () => {
...
-export class NzTestTimePickerComponent {
+export class NzTestTimePickerComponent implements OnInit {
   open = false;
   openChange = jasmine.createSpy('open change');
   autoFocus = false;
   date: Date | string = new Date();
   disabled = false;
   use12Hours = false;
   nzSuffixIcon?: string;
+  i18nPlaceHolder$: Observable<string | undefined> = of(undefined);
```
Comment (wenqi73) [inline components/time-picker/time-picker.component.spec.ts:206]:
```suggestion

```

File: components/time-picker/time-picker.component.spec.ts [hunk 4]
```diff
@@ -166,20 +188,28 @@ describe('time-picker', () => {
       (ngModelChange)="onChange($event)"
       [(nzOpen)]="open"
       (nzOpenChange)="openChange($event)"
+      [nzPlaceHolder]="i18nPlaceHolder$ | async"
```
Comment (wenqi73) [inline components/time-picker/time-picker.component.spec.ts:191]:
```suggestion
```

File: components/time-picker/time-picker.component.spec.ts [hunk 5]
```diff
@@ -151,6 +153,26 @@ describe('time-picker', () => {
...
+
+      beforeEach(inject([NzI18nService], (s: NzI18nService) => {
+        srv = s;
+      }));
+
+      it('should detect the language changes', fakeAsync(() => {
+        let placeHolderValue: string | undefined;
+        testComponent.i18nPlaceHolder$.subscribe(v => (placeHolderValue = v));
+
+        expect(placeHolderValue).toBe('请选择时间');
```
Comment (wenqi73) [inline components/time-picker/time-picker.component.spec.ts:167]:
```
placeholder = timeElement.nativeElement.querySelector('input').placeholder;
```

**Comment by wenqi73** [comment]
@DepickereSven The formatting change is OK, no need to change this.

**Review by wenqi73 — APPROVED** [review]
Thanks @DepickereSven !

**Comment by DepickereSven** [comment]
@wenqi73 You're welcome and thank you for reviewing!

**Pull request merged**